### DISCUSSION TITLE
[4.4.0-beta.6]: Browser history not updated when browsing asset subfolders

### DISCUSSION THREAD

**Discussion by thupsi** [craftcms/cms#12826] [Ideas]
### Description
When the user browses subfolders in any asset index, the url is updated but the browser history is not, so the user loses among other things the ability to navigate "back" to the last folder view.

### Steps to reproduce

1. Go to an asset volume and navigate to any subfolder. 
2. Try to go back to the last folder view using native browser functionality.

### Expected behavior

Go to the last viewed folder.

### Actual behavior

The CP displays the last page visited before "assets".

### Craft CMS version

4.4.0-beta.6

### Installed plugins and versions

-

**Comment by thupsi** [comment]
Ok, just checked and realized that this was also the case with the previous version of volume subfolders. 

Thing is, now that the folder tree is not visible at all times, the user has to move back and forth to go to sibling folders, so the native browser "back" functionality becomes more important.

I can move to a discussion if you don't consider this a bug (I guess technically it isn't).

**Reply by brandonkelly** [reply]
This is currently working as intended. Open to changing it, but if we start supporting Back/Forward buttons for asset source paths, we should also supporting them for sidebar sources as well (which are an even larger context change than navigating between subfolders).

**Reply by piotrpog** [reply]
Would be great if you changed it then, its pretty annoying to keep ending up in dashboard when i instinctively press back button to go back to previous directory :)

**Reply by thupsi** [reply]
With the new assets UI the user is far more likely to use the back button, but I do think it makes sense to make the change in both places (assets and sidebar sources).

**Reply by mikesnoeren** [reply]
Building on the previous discussions, it's clear that the current behavior of the asset navigation is not intuitive and is causing frustration among users. The expectation, especially with the URL updating to reflect the current folder, is that the browser's back functionality would allow users to easily navigate back to the previous folder.

The removal of the folder tree view and the lack of a quick back navigation option have significantly impacted the user experience, making it more challenging to navigate through deeply nested structures. Yes, you have breadcrumbs available, but those don't offer the same level of flexibility as a full tree navigation does (and only allow for navigating backwards). This has been echoed by multiple clients who have expressed a desire to revert the asset page to its previous state.

The makers of CKEditor have created a good alternative for the asset solution named CKBox. Seeing as CKEditor is integrated so closely into Craft, maybe that's something to look into? CKEditor and CKBox also work together very nicely.
